### DISCUSSION TITLE
Is there a way to list all variables of vars context?

### DISCUSSION THREAD

**Discussion by ehg-mengxu** [community/community#55813] [Actions] [answered]
### Select Topic Area

Question

### Body

I'm trying to find a way to be able to list all variables of vars context for an environment, as use a filter to get a subset of the variables to be piped into a configuration file. In order to make this function for different applications, it would be good to make the action agnostic about the variable names, however a prefix will be applied to variable names to work with the filter.

I've been searching for such solution in GitHub action documents, but no luck, so hope I can find answer here. Much appreciated.

**Comment by hrishikesh-kocharekar-dev** [comment]
To list all variables of the `vars` context for an environment and filter them based on a prefix, you can use the following approach in a GitHub Actions workflow:
name: List and Filter Variables

on:
  push:
    branches:
      - main

jobs:
  listAndFilterVariables:
    runs-on: ubuntu-latest
    steps:
      - name: List Variables
        run: |
          echo "Variables:"
          env | grep -e '^VAR_PREFIX_'

The workflow is triggered when there is a push to the `main` branch.
The job `listAndFilterVariables` runs on an Ubuntu environment.
In the `steps` section, the step named "List Variables" is executed.
The `run` field of the step contains a shell command that echoes the variables and filters them based on a prefix. You need to replace `VAR_PREFIX_` with your desired prefix. The `env` command lists all environment variables, and the `grep` command filters the output based on lines that start with the specified prefix.
You can customize this workflow to suit your needs. For example, you can add additional steps to manipulate the filtered variables or pipe them into a configuration file.

Remember to replace `main` with the appropriate branch name in the `on.push.branches` section if you want to trigger the workflow for a different branch.

**Comment by airtower-luna** [comment, marked as answer]
You can use [`toJSON()`](https://docs.github.com/en/actions/learn-github-actions/expressions#tojson) to get a JSON representation of any Actions object (see also: [printing context information to the log](https://docs.github.com/en/actions/learn-github-actions/contexts#example-printing-context-information-to-the-log)):

```yaml
      - name: Dump vars context
        env:
          VARS_JSON: ${{ toJSON(vars) }}
        run: echo "${VARS_JSON}"
```

You could redirect the output to a file, or process it with `jq`, or any other tool (including a custom script).
